allure-generator/src/main/javascript/utils/parseAddress.js

```diff
@@ -0,0 +1,60 @@
+const RFC2822_ADDRESS = /^(.*) <(.*)>$/;
+const LOOKS_LIKE_EMAIL = /^[^@]+@[^@]+$/;
+
+/**
+ * Parse a potentially RFC 2822 address into a display name and an address.
+ *
+ * @param {string | null | undefined} maybeAddress
+ * @returns {{ displayName: string, url?: string } | null
+ */
+export default function parseAddress(maybeAddress) {
```

Comment:
Moving this code to the generator would be nice instead of parsing the address during render. See https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/java/io/qameta/allure/owner/OwnerPlugin.java

Reply:
Okay, so I just need to accomplish the job on Java side and pass { displayName, url } to Backbone to render, right?

As for the overall idea, does it look fine?

Reply:
> As for the overall idea, does it look fine?

it is reasonable I would say. 


> Okay, so I just need to accomplish the job on Java side and pass { displayName, url } to Backbone to render, right?

[This line](https://github.com/allure-framework/allure2/blob/main/allure-generator/src/main/java/io/qameta/allure/owner/OwnerPlugin.java#L47) adds extra. This is how to access it:

<img width="335" alt="Screenshot 2024-03-12 at 17 45 45" src="https://github.com/allure-framework/allure2/assets/2149631/dd910dbd-6394-4609-b6e9-b43ffb4a4b0a">